utils/ixdtf/src/parsers/records.rs

```diff
@@ -108,62 +114,36 @@ pub struct DurationParseRecord {
     /// The `days` value.
     pub days: i32,
     /// The `hours` value.
-    pub hours: f64,
+    pub hours: i32,
     /// The `minutes` value.
-    pub minutes: f64,
+    pub minutes: i32,
     /// The `seconds` value.
-    pub seconds: f64,
-    /// The `milliseconds` value.
-    pub milliseconds: f64,
-    /// The `microseconds` value.
-    pub microseconds: f64,
-    /// The `nanoseconds` value.
-    pub nanoseconds: f64,
+    pub seconds: i32,
+    /// Any fraction part of a duration.
+    pub fraction: Option<DurationFraction>,
 }
 
+/// An enum representing the fraction part of a duration.
+#[non_exhaustive]
 #[cfg(feature = "duration")]
-impl DurationParseRecord {
-    pub(crate) fn from_records(
-        sign: bool,
-        date: DateDurationRecord,
-        time: TimeDurationRecord,
-    ) -> Self {
-        let minutes = if time.fhours > 0.0 {
-            time.fhours * 60.0
-        } else {
-            f64::from(time.minutes)
-        };
-
-        let seconds = if time.fminutes > 0.0 {
-            time.fminutes * 60.0
-        } else if time.seconds > 0 {
-            f64::from(time.seconds)
-        } else {
-            (minutes % 1.0) * 60.0
-        };
-
-        let milliseconds = if time.fseconds > 0.0 {
-            time.fseconds * 1000.0
-        } else {
-            (seconds % 1.0) * 1000.0
-        };
-
-        let microseconds = (milliseconds % 1.0) * 1000.0;
-        let nanoseconds = (microseconds % 1.0) * 1000.0;
-
-        let sign: i32 = if sign { 1 } else { -1 };
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum DurationFraction {
+    /// The fraction value applied to an hour.
+    Hours(i64),
```

Comment:
Question: why is this signed?

Reply:
Does ISO-8601 allow mixed signed duration strings? I think the answer is No, in which case you should make all the fields unsigned and introduce a single sign field, similar to UtfOffsetRecord.

Reply:
Basically my guiding principle is that these records should encode as many invariants as possible. If it isn't possible for there to be mixed signs, then the record should reflect that in its data types.